### DISCUSSION TITLE
Converting old hubot external scripts from coffeescript to mjs?

### DISCUSSION THREAD

**Discussion by yincrash** [hubotio/hubot#1723] [Q&A] [answered]
I'm trying to restart from an extremely old version of hubot and have a few coffee scripts. Is there any guide to migrating or can I run them through a converter of some sort?

**Comment by joeyguerra** [comment, marked as answer]
I haven't created a migration guide. My current thought is that you can use [decaffeinate](https://decaffeinate-project.org) (or ChatGP, or Co-Pilot) to convert the scripts to Javascript.

If you're willing to iterate on it, you can post questions here and I'll answer as best I can to help with the migration. What are your thoughts?

**Reply by yincrash** [reply]
Okay, so I ended up using [coffeescript-to-typescript](https://github.com/jaredp/coffeescript-to-typescript) to convert to TypeScript, and got it to run, however it appears that `User.get()`/`set()` now requires a datastore, however it's not clear how to initialize the `InMemoryDatastore`.

**Reply by yincrash** [reply]
So, I'm rewriting to use `robot.brain.users()` instead of `robot.users()` to use the existing brain interface that my script was using before. The `@types/hubot` package has `robot.datastore` set to readonly, so if other scripts want to use the datastore and TypeScript, they would need to alter the `@types/hubot` package.

**Reply by joeyguerra** [reply]
Here's the [DataStore test](https://github.com/hubotio/hubot/blob/main/test/DataStore_test.mjs) that initializes an InMemoryStore.

**Reply by yincrash** [reply]
Thanks! Successfully migrated my trivia bot - https://github.com/yincrash/hubot-trivia-game
